core/optaplanner-constraint-drl/pom.xml

```diff
@@ -108,6 +108,10 @@
           </ignoredUnusedDeclaredDependencies>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>animal-sniffer-maven-plugin</artifactId>
+      </plugin>
```

Comment:
I'd expect this to fail, because Drools does not work on Android.

Reply:
That is why I didnt put it there.

Reply:
Yes, it fails on the drl I removed the check https://github.com/kiegroup/optaplanner/pull/2083/commits/76391ac84e021742e5700a1a4de99ed9ba8e0366
But not only, I expect to have failures in core-impl as well. At this stage I need you to review them and decide if we ignore it and remove android support or we are going to fix that?

Reply:
now you can see errors in core-impl
```
org.optaplanner:optaplanner-core-impl:jar:8.5.0.Final in central (https://repo.maven.apache.org/maven2). The API analysis will not proceed.
2022-07-19T09:16:54.9473452Z [INFO] API checks completed without failures.
2022-07-19T09:16:54.9555751Z [INFO] 
2022-07-19T09:16:54.9558905Z [INFO] --- animal-sniffer-maven-plugin:1.20:check (animal-sniffer) @ optaplanner-core-impl ---
2022-07-19T09:16:56.0175126Z [INFO] Checking unresolved references to net.sf.androidscents.signature:android-api-level-32:12_r1
2022-07-19T09:16:56.9946695Z [ERROR] /home/runner/work/optaplanner/optaplanner/kiegroup_optaplanner/optaplanner/core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/common/accessor/LambdaBeanPropertyMemberAccessor.java:57: Undefined reference: java.lang.invoke.CallSite java.lang.invoke.LambdaMetafactory.metafactory(java.lang.invoke.MethodHandles.Lookup, String, java.lang.invoke.MethodType, java.lang.invoke.MethodType, java.lang.invoke.MethodHandle, java.lang.invoke.MethodType)
2022-07-19T09:16:56.9954697Z [ERROR] /home/runner/work/optaplanner/optaplanner/kiegroup_optaplanner/optaplanner/core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/common/accessor/LambdaBeanPropertyMemberAccessor.java:70: Undefined reference: java.util.function.Function java.lang.invoke.MethodHandle.invokeExact()
2022-07-19T09:16:56.9963028Z [ERROR] /home/runner/work/optaplanner/optaplanner/kiegroup_optaplanner/optaplanner/core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/common/accessor/LambdaBeanPropertyMemberAccessor.java:83: Undefined reference: java.lang.invoke.CallSite java.lang.invoke.LambdaMetafactory.metafactory(java.lang.invoke.MethodHandles.Lookup, String, java.lang.invoke.MethodType, java.lang.invoke.MethodType, java.lang.invoke.MethodHandle, java.lang.invoke.MethodType)
2022-07-19T09:16:56.9970146Z [ERROR] /home/runner/work/optaplanner/optaplanner/kiegroup_optaplanner/optaplanner/core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/common/accessor/LambdaBeanPropertyMemberAccessor.java:93: Undefined reference: java.util.function.BiConsumer java.lang.invoke.MethodHandle.invokeExact()
```

Reply:
See https://stackoverflow.com/questions/47783672/cant-find-referenced-class-java-lang-invoke-lambdametafactory

Reply:
Do you mean that it could be avoided my ignoring the warning?

Reply:
Yes.